### ISSUE TITLE
Dealing with disabled binding is confusing

### ISSUE THREAD

**Issue by jelbourn** [angular/angular#1162]
Say I have a component that needs to know about its disabled state in some event:
```js
@Component({
  selector: '[md-button][href]',
  bind: {disabled: 'disabled'}
})
class MdAnchor {
  onClick(event) {
    // Disabled anchors don't navigate anywhere.
    if (this.disabled) {
      event.preventDefault();
    }
  }
}
```

This will work fine if we have something like
```html
<a md-button href="http://google.com" [disabled]="isDisabled()">Google</a>
```

However, if you instead do
```html
<a md-button href="http://google.com" disabled>Google</a>
```
This code will not work, as `this.disabled` will be the empty string. I would have to update the value in `onChange`:
```js
onChange() {
  this.disabled = this.disabled || this.disabled == '';
}
```
or always use a method:
```js
isDisabled() {
  return this.disabled || this.disabled == '';
}
```

TL;DR: Change detection will set disabled to the empty string, which is falsy, but still implies disabled state.

**Comment by matanlurey** [comment]
I was using the following pattern in some experiments (Dart below):

```dart
bool _disabled = false;
void set disabled(disabled) {
  _disabled = disabled == false;
}
```

**Comment by mhevery** [comment]
close this one as a dupe of #1162

**Comment by zoechi** [comment]
@mhevery this is now a dup of itself. Perhaps wrong issue number pasted?
